src/components/facet-browser/FacetBrowserDisclosure.tsx

```diff
@@ -0,0 +1,87 @@
+import ExpandMoreIcon from "@danskernesdigitalebibliotek/dpl-design-system/build/icons/collection/ExpandMore.svg";
+import clsx from "clsx";
+import React, { FC, ReactNode, useCallback } from "react";
+import { useText } from "../../core/utils/text";
+import Pagefold from "../pagefold/Pagefold";
+
+export interface FacetBrowserDisclosureProps {
+  id: string;
+  mainIconPath?: string;
+  title: string;
+  children?: ReactNode;
+  isAvailable?: boolean;
+  fullWidth?: boolean;
+  showContent?: boolean;
+  removeHeadlinePadding?: boolean;
+  onClick?: () => void;
+}
+
+const FacetBrowserDisclosure: FC<FacetBrowserDisclosureProps> = ({
+  id,
+  title,
+  children,
+  mainIconPath,
+  isAvailable,
+  fullWidth,
+  showContent = false,
+  removeHeadlinePadding,
+  onClick
+}) => {
+  const t = useText();
+  const onClickHandler = useCallback(() => {
+    if (onClick) {
+      onClick();
+    }
+  }, [onClick]);
+
+  const disclosureId = `facet-${id}`;
+
+  return (
+    <div
+      className={`disclosure text-body-large ${
+        fullWidth ? "disclosure--full-width" : ""
+      }`}
+      role="button"
+      aria-controls={disclosureId}
+      aria-expanded={showContent}
+    >
+      <div
+        className={clsx(
+          "disclosure__headline text-body-large",
+          removeHeadlinePadding && "disclosure__headline--no-padding"
+        )}
+        onClick={onClickHandler}
+        onKeyDown={onClickHandler}
+        role="button"
+        tabIndex={0}
+      >
```

Comment:
Please rework this structure.

It renders two nested elements each with `role="button"`. That does not seem right. I would expect the headline to be the button and then it controls the element with the children.